apps/s_server.c

```diff
@@ -511,21 +513,31 @@ static int get_ocsp_resp_from_responder(SSL *s, tlsextstatusctx *srctx,
     proxy = srctx->proxy;
     no_proxy = srctx->no_proxy;
 
-    inctx = X509_STORE_CTX_new();
-    if (inctx == NULL)
+    ssl_ctx = SSL_get_SSL_CTX(s);
+    if (!SSL_CTX_get0_chain_certs(ssl_ctx, &chain))
         goto err;
-    if (!X509_STORE_CTX_init(inctx,
-                             SSL_CTX_get_cert_store(SSL_get_SSL_CTX(s)),
-                             NULL, NULL))
-        goto err;
-    obj = X509_STORE_CTX_get_obj_by_subject(inctx, X509_LU_X509,
-                                            X509_get_issuer_name(x));
-    if (obj == NULL) {
-        BIO_puts(bio_err, "cert_status: Can't retrieve issuer certificate.\n");
-        goto done;
+    if (chain) {
+        /* use untrusted server certificate chain (-cert_chain option) */
+        issuer = sk_X509_value(chain, 0);
+        if (!X509_name_cmp(X509_get_issuer_name(x), X509_get_subject_name(issuer)))
```

Comment:
Please use `X509_name_cmp() == 0`. We use boolean ops only on true/false return values.